collect_app/src/main/java/org/odk/collect/android/activities/FirstLaunchActivity.kt

```diff
@@ -59,17 +98,42 @@ class FirstLaunchActivity : LocalizedActivity() {
                 versionInformation.versionToDisplay
             )
 
-            configureLater.addOnClickListener {
-                Analytics.log(AnalyticsEvents.TRY_DEMO)
+            dontHaveServer.apply {
+                text = SpannableStringBuilder()
+                    .append(getString(org.odk.collect.strings.R.string.dont_have_project))
+                    .append(" ")
+                    .color(getThemeAttributeValue(context, com.google.android.material.R.attr.colorAccent)) {
+                        append(getString(org.odk.collect.strings.R.string.try_demo))
+                    }
 
+                setOnClickListener {
+                    viewModel.tryDemo()
+                }
+            }
+        }
+    }
+}
+
+private class FirstLaunchViewModel(
+    private val scheduler: Scheduler,
+    private val projectsRepository: ProjectsRepository,
+    private val projectsDataService: ProjectsDataService
+) : ViewModel() {
+    private val _isLoading = MutableLiveData<Boolean>()
```

Comment:
Why make this nullable?

Reply:
Because I don't want to throw false as the initial value. That would mess up what we have above in the activity class plus when it comes to the progress dialog it also does not need to receive that initial false value.
And one more thing in kotlin it's not nullable because the data type is `Boolean` not `Boolean?`

Reply:
Ah because you need to differentiate between "no longer loaded" and "hasn't loaded yet" for the observation that launches the Main Menu. It's strange because the `LiveData` is technically nullable (you could set it to `null`) due to some weirdness in the generic typing which was the whole reason to introduce the `NonNull...` implementations. 

Here I guess this all works without a null check (again in the observation that launches the Main Menu) because `_isLoading` has no "initial" value. However, I think if you called `getValue` on it, you would get back `null`. Given that, I'd prefer to use `MutableLiveData<Boolean?>(null)` and then add a null check in so that it's clearer what's happening. It looks like the newest `lifecycle` modules actually give us [correct typing for nullable types](https://developer.android.com/jetpack/androidx/releases/lifecycle#2.6.0).

Reply:
I already added a null check in MaterialProgressDialogFragment because it's java but you think it would be good to do the same here:
```
            viewModel.isLoading.observe(this@FirstLaunchActivity) { isLoading ->
                if (!isLoading) {
                    ActivityUtils.startActivityAndCloseAllOthers(
                        this@FirstLaunchActivity,
                        MainMenuActivity::class.java
                    )
                }
            }
```

when observing it can't be null you said that it's possible to get null by calling `getValue() ` but if I'm not using that method I don't think if it makes sense to change anything.

Reply:
Ok I realized I was being over cautious because of how `LiveData` **used** to work. I thought I'd be able to write the following line in `FirstLaunchViewModel`:

```kotlin
_isLoading.value = null
```

This used to be possible, but it is now fixed (a lint check fails). I knew they'd improved this, but hadn't realised we'd finally got protection on the setter side as well as in the `Observer`.

I think this means that we can actually deprecate our custom `NonNull` types!